substrate/client/rpc-spec-v2/src/common/events.rs

```diff
@@ -136,17 +136,221 @@ pub struct ArchiveStorageMethodOk {
 }
 
 /// The error of a storage call.
-#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
 #[serde(rename_all = "camelCase")]
 pub struct ArchiveStorageMethodErr {
 	/// Reported error.
 	pub error: String,
 }
 
+/// The type of the archive storage difference query.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub enum ArchiveStorageDiffType {
+	/// The result is provided as value of the key.
+	Value,
+	/// The result the hash of the value of the key.
+	Hash,
+}
+
+/// The storage item to query.
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ArchiveStorageDiffItem<Key> {
+	/// The provided key.
+	pub key: Key,
+	/// The type of the storage query.
+	pub return_type: ArchiveStorageDiffType,
+	/// The child trie key if provided.
+	#[serde(skip_serializing_if = "Option::is_none")]
+	#[serde(default)]
+	pub child_trie_key: Option<Key>,
+}
+
+/// The result of a storage difference call.
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ArchiveStorageDiffMethodResult {
+	/// Reported results.
+	pub result: Vec<ArchiveStorageDiffResult>,
+}
+
+/// The result of a storage difference call operation type.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub enum ArchiveStorageDiffOperationType {
+	/// The key is added.
+	Added,
+	/// The key is modified.
+	Modified,
+	/// The key is removed.
+	Deleted,
+}
+
+/// The result of an individual storage difference key.
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ArchiveStorageDiffResult {
+	/// The hex-encoded key of the result.
+	pub key: String,
+	/// The result of the query.
+	#[serde(flatten)]
+	pub result: StorageResultType,
+	/// The operation type.
+	#[serde(rename = "type")]
+	pub operation_type: ArchiveStorageDiffOperationType,
+	/// The child trie key if provided.
+	#[serde(skip_serializing_if = "Option::is_none")]
+	#[serde(default)]
+	pub child_trie_key: Option<String>,
+}
+
+/// The event generated by the `archive_storageDiff` method.
+///
+/// The `archive_storageDiff` can generate the following events:
+///  - `storageDiff` event - generated when a `ArchiveStorageDiffResult` is produced.
+///  - `storageDiffError` event - generated when an error is produced.
+///  - `storageDiffDone` event - generated when the `archive_storageDiff` method completed.
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+#[serde(tag = "event")]
+pub enum ArchiveStorageDiffEvent {
+	/// The `storageDiff` event.
+	StorageDiff(ArchiveStorageDiffResult),
```

Comment:
```suggestion
	Ok(ArchiveStorageDiffResult),
```

Reply:
These variants will be serialized over the wire for subscriptions, I think it is ok to change them but will have to reflect in the archive PR as well 🤔 
For clarity do you think its better to change it from `"event": "storageDiff"` into `"event": "ok"`? Maybe we can change it to be plainly `"event": "diff"` 

https://github.com/paritytech/json-rpc-interface-spec/pull/161/files#diff-9fbae47d3367a2ec6feb5bee7a45aa5cbef704437c824b011a7f5b5117e7f922R54

Reply:
Yeah, it's fine my only point was that it was big annoying to repeat `Diff` both in enum name and variant.

Just ignore my comment :)